### ISSUE TITLE
 Limit the growth of the database

### ISSUE THREAD

**Issue by damianperez** [php-telegram-bot/core#1003]
### 🎉 Feature Request
Limit the growth of the database
#### Summary
Setting the days to save in logs of the tables in webhook

If the bot exceeds a number of users that any server service can handle, the tables begin to grow exponentially. It would be good if (for example, the request_limiter table) does not store more than what is previously configured in the bot initialization. With thousands of records in the db, cleanup is almost impossible.
A possible solution would be to find an appropriate key in the tables so that you can do:
insert ignore into .... on duplicate key update ...
If not, compare the preset number of records in initialization with the amount in the table and clean them automatically.

**Comment by lawrencegs** [comment]
Yes please, my `message` table have 35 million+ rows & the /cleanup took hours...

**Comment by Alexell** [comment]
I join the problem. The bot stores a lot of data in the database and cleaning with `/clean` takes several hours, during which the bot slows down a lot. I wanted to do an automatic cleaning of the database on a schedule, but due to the fact that `FOREIGN KEY` is present in the tables, it is impossible to clean the tables with my usual queries, so I don’t know what to do.
